src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEngine.Unix.cs

```diff
@@ -157,20 +157,10 @@ private SocketAsyncEngine()
                     throw new InternalException(err);
                 }
 
-                bool suppressFlow = !ExecutionContext.IsFlowSuppressed();
-                try
-                {
-                    if (suppressFlow) ExecutionContext.SuppressFlow();
-
-                    Thread thread = new Thread(s => ((SocketAsyncEngine)s!).EventLoop());
-                    thread.IsBackground = true;
-                    thread.Name = ".NET Sockets";
-                    thread.Start(this);
-                }
-                finally
-                {
-                    if (suppressFlow) ExecutionContext.RestoreFlow();
-                }
+                var thread = new Thread(static s => ((SocketAsyncEngine)s!).EventLoop());
```

Comment:
Probably worth looking to see if there are other threads that need to do this. Like the FileSystemWatcher threads.

Reply:
I did an audit of all threads we're creating; these were the only ones that looked appropriate.  The FileSystemWatcher ones aren't appropriate as they raise user events; it's only fine for macOS (which I changed) as ExecutionContext is flown manually there.

Reply:
Then we're gonna need an overload to disable it over there as it captures the context. I'll fill another bug. Anything created lazily is notorious for capturing the wrong execution context

Reply:
You already opened https://github.com/dotnet/runtime/issues/30954 in September.

Reply:
Ah yes